Refactored some time-related assertions to introduce a tolerance rang…

Vincenzo-Massaro opened the pull request:
[QA-5439](https://pagopa.atlassian.net/browse/QA-5439)

[QA-5439]: https://pagopa.atlassian.net/browse/QA-5439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Vincenzo-Massaro committed:
Refactored some time-related assertions to introduce a tolerance range in the comparison

Vincenzo-Massaro committed:
Marked all QA5439 related tests

Vincenzo-Massaro committed:
Correct timestamps comparison and IUN logs

- Two of the 'isCloseTo' assertions introduced in AdvanceNotificationsB2bSteps.java compared wrong pairs of timestamps
- Introduced handling of AssertionError in SharedSteps.java so that the IUN can be logged also if this exception is thrown

Vincenzo-Massaro commented on src/test/java/it/pagopa/pn/cucumber/QA5439Test.java:
E' una suite temporanea per facilitare il test dei casi coinvolti, si intende rimuoverla prima del merge, insieme ai tag posti sui relativi scenari Cucumber

Vincenzo-Massaro commented on src/test/java/it/pagopa/pn/cucumber/steps/SharedSteps.java:
```diff
@@ -1987,11 +2030,20 @@ public String getGherkinIrreperibileTaxId() {
         throw new AssertionFailedError(message, assertionFailedError.getExpected(), assertionFailedError.getActual(), assertionFailedError.getCause());
     }
 
+    public void throwAssertFailerWithIUN(AssertionError assertionError) {
```
E' stata aggiunta la stessa gestione per il tipo AssertionError, lanciato dal framework AssertJ al posto di AssertionFailedError, lanciato da plain JUnit

Vincenzo-Massaro commented on src/test/java/it/pagopa/pn/cucumber/steps/pa/AvanzamentoNotificheB2bSteps.java:
```diff
@@ -577,32 +622,39 @@ public void verificationDateComplettelyUnreachableWithRefinement(Integer destina
-            Assertions.assertEquals(shedulingDate, prepareAnalogDomicileFailureTimestamp);
-            Assertions.assertEquals(sendAnalogProgressTimestampDate, sendAnalogProgressNotificationDate);
-            Assertions.assertEquals(sendFeedbackTimestampDate, sendFeedbackNotificationDate);
+            assertThat(completelyUnreachableDate)
```
Introdotto range di tolleranza per il confronto di timestamps, posto a 1 secondo come suggerito da @EmanueleBonofiglio

Vincenzo-Massaro commented on src/test/java/it/pagopa/pn/cucumber/steps/pa/AvanzamentoNotificheB2bSteps.java:
```diff
@@ -577,32 +622,39 @@ public void verificationDateComplettelyUnreachableWithRefinement(Integer destina
-        } catch (AssertionFailedError assertionFailedError) {
-            sharedSteps.throwAssertFailerWithIUN(assertionFailedError);
+            // Assertions.Assertions.assertEquals(sendFeedbackDate,sendAnalogProgressDate);
+        } catch (AssertionError assertionError) {
```
Le assertions "isCloseTo" lanciano AssertionError al posto di AssertionFailedError

Vincenzo-Massaro committed:
Revert "Marked all QA5439 related tests"

This reverts commit ca47aa6bb3b95231a22947ef439911021881bb90.

mmancini95 commented on src/test/java/it/pagopa/pn/cucumber/steps/pa/AvanzamentoNotificheB2bSteps.java:
```diff
@@ -2606,7 +2660,7 @@ public void vieneSchedulatoIlPerfezionamento(String timelineCategory, @Transpose
         System.out.println(schedulingDate.format(fmt1));
         Assertions.assertEquals(timelineElement.getDetails().getSchedulingDate().format(fmt1), schedulingDate.format(fmt1));
-        //Assertions.assertEquals(timelineElement.getDetails().getSchedulingDate(), schedulingDate);
+        //Assertions.Assertions.assertEquals(timelineElement.getDetails().getSchedulingDate(), schedulingDate);
```
**nitpick:** Can we completely remove this commented line?

Vincenzo-Massaro replied:
fatto 👍

mmancini95 commented on src/test/java/it/pagopa/pn/cucumber/steps/pa/AvanzamentoNotificheB2bSteps.java:
```diff
@@ -577,32 +622,39 @@ public void verificationDateComplettelyUnreachableWithRefinement(Integer destina
-            // Assertions.assertEquals(sendFeedbackDate,sendAnalogProgressDate);
-        } catch (AssertionFailedError assertionFailedError) {
-            sharedSteps.throwAssertFailerWithIUN(assertionFailedError);
+            // Assertions.Assertions.assertEquals(sendFeedbackDate,sendAnalogProgressDate);
```
same here

Vincenzo-Massaro replied:
e rifatto 👍

Vincenzo-Massaro committed:
Removed some outdated commented lines

Vincenzo-Massaro committed:
Merge remote-tracking branch 'origin/develop' into feature/QA-5439-Precisione-timestamp-eccessiva-in-alcuni-test

Vincenzo-Massaro committed:
Merge remote-tracking branch 'origin/develop' into feature/QA-5439-Precisione-timestamp-eccessiva-in-alcuni-test

# Conflicts:
#	src/test/java/it/pagopa/pn/cucumber/steps/pa/AvanzamentoNotificheB2bSteps.java

Vincenzo-Massaro committed:
Fixed 1 missing import of AvanzamentoNotificheB2bSteps.java

Vincenzo-Massaro committed:
Removed CON020Sequence890Test.java and related tags

The CON020Sequence890Test suite was only useful for the development of the https://pagopa.atlassian.net/browse/QA-4604 ticket, once it was finished there was no reason to keep it.
